src/input.ts

```diff
@@ -7,14 +7,19 @@ export type ZipFileDescription = {
   bytes: ReadableStream<Uint8Array> | Uint8Array | Promise<Uint8Array>
   crc?: number // will be computed later
 }
+export type ZipFolderDescription = {
+  modDate: Date
+  bytes?: undefined
+  crc?: undefined
+}
```

Comment:
The fact that the CRC remains undefined is more a shortcut than proper semantics (it should be set to zero, we're lucky that passing undefined to setUint32 also works). So `ZipFolderDescription` is really just `ZipFileDescription` without bytes.

Reply:
Unfortunately TypeScript is not that clever. This was hack indeed to go around it without need to refactor it too much. I improved it by adding isFile to both which is nice trade off I think and now it works just fine.